### PR TITLE
RHIDP-3111 Replace references to deprecated backend.auth.keys config

### PR DESCRIPTION
**Pull request by themr0c**
**IMPORTANT: Do Not Merge - To be merged by Docs Team Only**

**Version(s):** 1.3.1

**Issue:** https://issues.redhat.com/browse/RHIDP-3111

**Link to docs preview:**

**Reviews:**
- [ ] SME: @ mention assignee
- [ ] QE: @albarbaro
- [ ] Docs review: @ mention assignee
- [ ] Additional review: @mention assignee (by writer)

**Additional information:**

### CODE REVIEW

File: modules/installation/proc-add-custom-app-config-file-ocp-operator.adoc [hunk 1]
```diff
@@ -34,7 +34,7 @@ data:
       baseUrl: <{product-very-short}_URL> # <1>
     backend:
       auth:
-        keys:
+        externalAccess:
           - secret: "${BACKEND_SECRET}" # <2>
```
Comment (04kash) [inline modules/installation/proc-add-custom-app-config-file-ocp-operator.adoc:38]:
```suggestion
        externalAccess:
            - type: legacy
              options:
                subject: legacy-default-config
                secret: "${BACKEND_SECRET}"
```

File: modules/installation/proc-deploy-rhdh-instance-eks.adoc [hunk 2]
```diff
@@ -33,7 +33,7 @@ data:
       baseUrl: https://<rhdh_dns_name>
     backend:
       auth:
-        keys:
+        externalAccess:
           - secret: "${BACKEND_SECRET}"
```
Comment (04kash) [inline modules/installation/proc-deploy-rhdh-instance-eks.adoc:37]:
```suggestion
        externalAccess:
            - type: legacy
              options:
                subject: legacy-default-config
                secret: "${BACKEND_SECRET}"
```

File: modules/installation/proc-rhdh-deploy-aks-operator.adoc [hunk 3]
```diff
@@ -103,7 +103,7 @@ data:
       baseUrl: https://<app_address>
     backend:
       auth:
-        keys:
+        externalAccess:
           - secret: "${BACKEND_SECRET}"
```
Comment (04kash) [inline modules/installation/proc-rhdh-deploy-aks-operator.adoc:107]:
```suggestion
        externalAccess:
            - type: legacy
              options:
                subject: legacy-default-config
                secret: "${BACKEND_SECRET}"
```

**Review by 04kash — CHANGES_REQUESTED** [review]
Made a few changes. @rm3l could you take a look and see if this needs to be updated anywhere else?

**Review by rm3l — CHANGES_REQUESTED** [review]
> Made a few changes. @rm3l could you take a look and see if this needs to be updated anywhere else?

+1 to the review comments already written. I didn't see other files that would need to be updated other than the ones here.

**Review by albarbaro — COMMENTED** [review]
Agree on the changes already requested

File: modules/installation/proc-add-custom-app-config-file-ocp-operator.adoc [hunk 4]
```diff
@@ -34,8 +34,11 @@ data:
       baseUrl: <{product-very-short}_URL> # <1>
     backend:
       auth:
-        keys:
-          - secret: "${BACKEND_SECRET}" # <2>
+        externalAccess:
+            - type: legacy
+              options:
+                subject: legacy-default-config
+                secret: "${BACKEND_SECRET}"
```
Comment (themr0c) [inline modules/installation/proc-add-custom-app-config-file-ocp-operator.adoc:41]:
```suggestion
                secret: "${BACKEND_SECRET}" # <2>
```

**Review by hmanwani-rh — APPROVED** [review]

**Pull request merged**